cpu/stm32/periph/spi.c

```diff
@@ -240,13 +229,13 @@ void spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
     }
     uint8_t br = dividers[bus];
 
-    DEBUG("[spi] acquire: requested clock: %"PRIu32", resulting clock: %"PRIu32
-          " BR divider: %u\n",
+    DEBUG("[spi] acquire: requested clock: %" PRIu32
+         " Hz, resulting clock: %" PRIu32 " Hz, BR divider: %u\n",
           clk,
-          periph_apb_clk(spi_config[bus].apbbus)/(1 << (br + 1)),
-          br);
+          periph_apb_clk(spi_config[bus].apbbus) >> (br + 1),
+          (unsigned)br);
 
-    uint16_t cr1_settings = ((br << BR_SHIFT) | mode | SPI_CR1_MSTR);
+    uint16_t cr1_settings = ((br << BR_SHIFT) | mode | SPI_CR1_MSTR | SPI_CR1_SPE);
     /* Settings to add to CR2 in addition to SPI_CR2_SETTINGS */
     uint16_t cr2_extra_settings = 0;
```

Comment:
```suggestion
    uint16_t cr2_settings = SPI_CR2_SETTINGS;
```

Reply:
```suggestion
    uint16_t cr2 =  SPI_CR2_SETTINGS;
```

and cr1_settings can be cr1

Reply:
Done and squashed